src/main/java/sg/ncl/MainController.java

```diff
@@ -340,15 +345,20 @@ public String logout(HttpSession session) {
     
     @RequestMapping(value="/signup2", method=RequestMethod.GET)
     public String signup2(Model model) {
-    	// TODO get each model data and put into relevant ones
-    	model.addAttribute("loginForm", new LoginForm());
     	model.addAttribute("signUpMergedForm", new SignUpMergedForm());
     	return "signup2";
     }
     
     @RequestMapping(value="/signup2", method=RequestMethod.POST)
-    public String validateDetails(@ModelAttribute("loginForm") LoginForm loginForm, @ModelAttribute("signUpMergedForm") SignUpMergedForm signUpMergedForm) {
-    	// TODO get each model data and put into relevant ones
+    public String validateDetails(
+            @Valid
+            @ModelAttribute("signUpMergedForm") SignUpMergedForm signUpMergedForm,
+            BindingResult bindingResult,
+            final RedirectAttributes redirectAttributes) {
+
+        if (bindingResult.hasErrors() || signUpMergedForm.getIsValid() == false) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1125)